paasta_tools/kubernetes_tools.py

```diff
@@ -529,30 +529,44 @@ def get_autoscaling_metric_spec(
                     ),
                 )
             )
-        elif metrics_provider == "http":
+        elif metrics_provider in ("http", "uwsgi"):
             annotations = {"signalfx.com.custom.metrics": ""}
-            metrics.append(
-                V2beta1MetricSpec(
-                    type="Pods",
-                    pods=V2beta1PodsMetricSource(
-                        metric_name="http",
-                        target_average_value=target,
-                        selector=selector,
-                    ),
+            if (
+                autoscaling_params.get("forecast_policy") == "moving_average"
+                or "offset" in autoscaling_params
+            ):
+                metric_name = (
+                    f"{self.get_sanitised_deployment_name()}-{metrics_provider}"
                 )
-            )
-        elif metrics_provider == "uwsgi":
-            annotations = {"signalfx.com.custom.metrics": ""}
-            metrics.append(
-                V2beta1MetricSpec(
-                    type="Pods",
-                    pods=V2beta1PodsMetricSource(
-                        metric_name="uwsgi",
-                        target_average_value=target,
-                        selector=selector,
+                metrics.append(
+                    V2beta1MetricSpec(
+                        type="External",
+                        external=V2beta1ExternalMetricSource(
+                            metric_name=metric_name,
+                            target_value=target - autoscaling_params.get("offset", 0),
+                        ),
+                    )
+                )
+                signalflow = self.build_signalflow_for_autoscaling(
+                    metric_name=metrics_provider,
```

Comment:
I guess I was thinking that from the perspective of the method that builds that HPA spec, it's called the metrics provider because that's what we've been calling it in soa-configs.  But from the perspective of the the method that builds the signalflow, it's just the name of a SignalFX metric?

Reply:
oh i misunderstood you.  yes i will change the names a bit to make it clear that hpa metric name and signalfx metric name are different